examples/Fletcher_random_stream_performance/Fletcher_random_stream_performance.ino

```diff
@@ -16,166 +16,214 @@
 #else
 #define MAX_LEN 16384
 #endif
+union main_value_storage {
+  uint8_t uint8[MAX_LEN];
+  uint16_t uint16[MAX_LEN/2];
+  uint32_t uint32[MAX_LEN/4];
+} values;
 
 #define DO_N 23
 
-uint32_t myfletcher16(uint8_t *data, const uint16_t length)
-{
-  uint16_t s1 = 0;
-  uint16_t s2 = 0;
-  for (uint16_t i = 0; i < length; i++)
-  {
-    s1 = (s1 + data[i]) % ((((uint16_t) 1) << 8) - 1);
-    s2 = (s2 + s1) % ((((uint16_t) 1) << 8) - 1);
+#if defined(ARDUINO_ARCH_AVR)
+String float2strn(float value, size_t n) {
+  char strvalout[n];
+  dtostrf(value, n, 2, strvalout);
+  return strvalout;
+}
+#else
+String float2strn(float value, size_t n) {
+  String strval = String(value);
+  while (strval.length() < n) {
+    strval = " " + strval;
   }
-  return (s2 << 8) | s1;
+  return strval;
 }
+#endif
 
-uint32_t myfletcher32(uint16_t *data, const uint16_t length)
+uint16_t myfletcher16(uint8_t *data, const size_t length)
 {
-  uint32_t s1 = 0;
-  uint32_t s2 = 0;
-  for (uint16_t i = 0; i < length; i++)
+  uint8_t s1 = 0;
+  uint8_t s2 = 0;
+  for (size_t i = 0; i < length; i++)
   {
-    s1 = (s1 + data[i]) % ((((uint32_t) 1) << 16) - 1);
-    s2 = (s2 + s1) % ((((uint32_t) 1) << 16) - 1);
+    s1 = (uint8_t) (((uint16_t) (s1 + data[i])) % ((((uint16_t) 1) << 8) - ((uint16_t) 1)));
```

Comment:
possible overflow

Reply:
You are right with the overflow here. But storing the data before does not solve it.

My explicit casting was at the wrong point. Instead of
```c++
s1 = (uint32_t) (   ((uint64_t) (s1 + data[i]))    % ((((uint64_t) 1) << 32) - ((uint64_t) 1)));
s2 = (uint32_t) (   ((uint64_t) (s2 + s1))         % ((((uint64_t) 1) << 32) - ((uint64_t) 1)));
```
it should be:
```c++
s1 = (uint32_t) (   ((uint64_t) s1 + (uint64_t) data[i])    % ((((uint64_t) 1) << 32) - ((uint64_t) 1)));
s2 = (uint32_t) (   ((uint64_t) s2 + (uint64_t) s1)         % ((((uint64_t) 1) << 32) - ((uint64_t) 1)));
```

Or in your implementation instead of
```c++
s1 = s1 + data[i];
s2 = s2 + s1;
```
it should be:
```c++
s1 = ((uint64_t) s1) + ((uint64_t) data[i]);
s2 = ((uint64_t) s2) + ((uint64_t) s1);
```